subprojects/language-groovy/src/testFixtures/groovy/org/gradle/testing/fixture/GroovyCoverage.groovy

```diff
@@ -29,13 +31,23 @@ class GroovyCoverage {
     static final List<String> SUPPORTS_TIMESTAMP
     static final List<String> SUPPORTS_PARAMETERS
     static final List<String> SUPPORTS_DISABLING_AST_TRANSFORMATIONS
+    static final List<String> SINCE_3_0
+
+    /**
+     * The current Groovy version if stable, otherwise the latest stable version before the current version.
+     */
+    static final String CURRENT_STABLE
 
     static {
         SUPPORTED_BY_JDK = groovyVersionsSupportedByJdk(JavaVersion.current())
         SUPPORTS_GROOVYDOC = versionsAbove(SUPPORTED_BY_JDK, "1.6.9")
         SUPPORTS_TIMESTAMP = versionsAbove(SUPPORTED_BY_JDK, "2.4.6")
         SUPPORTS_PARAMETERS = versionsAbove(SUPPORTED_BY_JDK, "2.5.0")
         SUPPORTS_DISABLING_AST_TRANSFORMATIONS = versionsAbove(SUPPORTED_BY_JDK, "2.0.0")
+        SINCE_3_0 = versionsAbove(SUPPORTED_BY_JDK, "3.0.0")
+        CURRENT_STABLE =!isCurrentGroovyVersionStable()
```

Comment:
I think we shouldn't have the negation here. I also think the current branches are the wrong way round: If the version is stable, then we'd use the current version (`GroovySystem.version`, right?), if not, we'd use the lasted version which is below.

Reply:
Damn, yes, that's a typo.